### DISCUSSION TITLE
Implementing Structured Templates for Improved Project Management

### DISCUSSION THREAD

**Discussion by Niravanaa** [NathanGrenier/SOEN-341#35] [Ideas]
We can standardize how we structure the descriptions for commits, pull requests and to-do items/bug report items for the project. Templates can help streamline our workflow and ensure consistency and clarity across contributions from all members.

Here are some examples, subject to change if considered to be adopted:

# To-Do Task
![image](https://github.com/NathanGrenier/SOEN-341/assets/64038962/b98dec56-0b21-450f-9e03-54d2e613aeb3)

# Pull Requests
![image](https://github.com/NathanGrenier/SOEN-341/assets/64038962/4c19c330-62c6-4a21-ae1d-f71519d55ebb)

# Issues (Bug Reports)
![image](https://github.com/NathanGrenier/SOEN-341/assets/64038962/871abb61-8c93-46b8-897a-c95f904caf75)

# Commits
![image](https://github.com/NathanGrenier/SOEN-341/assets/64038962/75e32e9c-3c11-4369-a84f-713b293a82b9)

The markdown code for the templates can be further examined [here](https://pastebin.com/p80hZryp)

**Comment by NathanGrenier** [comment]
I like the idea. Templates for issues an PRs can be setup using [Github templates](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/about-issue-and-pull-request-templates).

I'll create an issue out of this discussion.

**Comment by CarciDev** [comment]
Yea good idea. Will help down the line.
